chain/ethereum/src/network_indexer/network_indexer.rs

```diff
@@ -0,0 +1,1540 @@
+                    "error" => format!("{}", e),
+                );
+
+                let state = state.take();
+
+                transition!(PollChainHead {
+                    local_head: state.local_head,
+                    prev_chain_head: Some(state.chain_head.into()),
+                    chain_head: poll_chain_head(context),
+                })
+            }
+        }
+    }
+
+    fn poll_vet_block<'a, 'c>(
+        state: &'a mut RentToOwn<'a, VetBlock>,
+        context: &'c mut RentToOwn<'c, Context>,
+    ) -> Poll<AfterVetBlock, Error> {
+        // Abort if the output stream has been closed.
+        try_ready!(context.event_sink.poll_ready());
+
+        let state = state.take();
+        let block = state.block;
+
+        // Validate the block.
+        if block.inner().number.is_none() || block.inner().hash.is_none() {
+            // This is fairly irregular, so log a warning.
+            warn!(
+                context.logger,
+                "Block number or hash missing; trying again";
+                "block" => format_block(&block),
+            );
+
+            // The block is invalid, throw away the entire stream and
+            // start with re-checking the chain head block again.
+            transition!(PollChainHead {
+                local_head: state.local_head,
+                prev_chain_head: Some(state.chain_head.into()),
+                chain_head: poll_chain_head(context),
+            })
+        }
+
+        // If we encounter a block that has a smaller number than our
+        // local head block, then we throw away the block stream and
+        // try to start over with a fresh chain head block.
+        //
+        // The assumption here is that maybe the network provider being
+        // used is temporarily serving from an outdated node.
+        let block_number = block.inner().number.unwrap().as_u64();
+        let local_head_number = state.local_head.map_or(0u64, |ptr| ptr.number);
+        if block_number < local_head_number {
+            // This is pretty irregular, so make log a warning.
+            warn!(
+                context.logger,
+                "Received older block than the local head; \
+                 re-evaluate chain head and try again";
+                "local_head" => state.local_head.map_or(
+                    String::from("none"), |ptr| format_block_pointer(&ptr)
+                ),
+                "block" => format_block(&block),
+            );
+
+            transition!(PollChainHead {
+                local_head: state.local_head,
+                prev_chain_head: Some(state.chain_head.into()),
+                chain_head: poll_chain_head(context),
+            })
+        }
+
+        // Check whether we have a reorg (parent of the new block != our local head).
+        if block.inner().parent_ptr() != state.local_head {
+            let depth = block.inner().number.unwrap().as_u64()
+                - state.local_head.map_or(0u64, |ptr| ptr.number);
```

Comment:
Eh, you're right, this will never report the _real_ depth of the reorg. I do want that information, but I think I can only log it once we have found the common ancestor.

Reply:
Updated